presto-spi/src/main/java/com/facebook/presto/spi/RecordCursor.java

```diff
@@ -43,6 +43,11 @@
 
     boolean isNull(int field);
 
+    default long getSystemMemoryUsage()
+    {
+        return 16 * 1024 * 1024; // GUESSED_MEMORY_USAGE: 16MB
```

Comment:
I think it's fine to make this guess in the Hive connector, but I don't think the SPI should make a guess like this. I would just return zero